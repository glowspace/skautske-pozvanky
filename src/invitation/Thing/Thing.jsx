import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { StandardText } from '../../styled/Styled';

const ThingsStyled = styled.div`
  max-width: 50%;
  display: flex;
  font-size: 1rem;
  display: flex;
  flex-flow: column wrap;
  max-height: 70%;
  justify-content: space-between;
  margin-bottom: 1.25rem;
  margin-left: 1rem;
  margin-right: 1rem;
`;

const Thing = ({ thing }) => {
  return (
    <ThingsStyled>
      <StandardText fontSize="1rem" color="rgb(154, 21, 143)">
        ‚òê {thing.name}
      </StandardText>
    </ThingsStyled>
  );
};

Thing.propTypes = {
  thing: PropTypes.shape({
    name: PropTypes.string
  })
};

Thing.defaultProps = {
  thing: {
    name: ''
  }
};

export default Thing;
