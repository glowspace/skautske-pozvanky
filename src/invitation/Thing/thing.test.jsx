import React from 'react';
import { render } from '@testing-library/react';
import Thing from './Thing';
import 'jest-styled-components';

describe('thing renders component', () => {
  test('renders correctly with no prop', () => {
    expect(render(<Thing />)).toMatchSnapshot();
  });

  test('renders correctly with thing as prop', () => {
    expect(
      render(
        <Thing
          thing={{
            name: 'test'
          }}
        />
      )
    ).toMatchSnapshot();
  });
  test('renders correctly with thing as prop in bw mode', () => {
    expect(
      render(
        <Thing
          thing={{
            name: 'test'
          }}
          bw
        />
      )
    ).toMatchSnapshot();
  });
});
