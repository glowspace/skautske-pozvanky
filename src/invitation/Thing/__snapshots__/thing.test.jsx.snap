// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thing renders component renders correctly with no prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  font-size: 1rem;
  color: #9a158f;
  font-style: normal;
}

.c0 {
  max-width: 50%;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: nowrap;
  -ms-flex-flow: nowrap;
  flex-flow: nowrap;
  max-height: 80%;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
          color="#9a158f"
          font-size="1rem"
        >
          ☐
        </div>
        <div
          class="c1"
          color="#9a158f"
          font-size="1rem"
          style="padding-left: 5px;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bxivhb cqFfly"
    >
      <div
        class="sc-htpNat kqfmGm"
        color="#9a158f"
        font-size="1rem"
      >
        ☐
      </div>
      <div
        class="sc-htpNat kqfmGm"
        color="#9a158f"
        font-size="1rem"
        style="padding-left: 5px;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`thing renders component renders correctly with thing as prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c1 {
  font-size: 1rem;
  color: #9a158f;
  font-style: normal;
}

.c0 {
  max-width: 50%;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: nowrap;
  -ms-flex-flow: nowrap;
  flex-flow: nowrap;
  max-height: 80%;
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
          color="#9a158f"
          font-size="1rem"
        >
          ☐
        </div>
        <div
          class="c1"
          color="#9a158f"
          font-size="1rem"
          style="padding-left: 5px;"
        >
          test
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bxivhb cqFfly"
    >
      <div
        class="sc-htpNat kqfmGm"
        color="#9a158f"
        font-size="1rem"
      >
        ☐
      </div>
      <div
        class="sc-htpNat kqfmGm"
        color="#9a158f"
        font-size="1rem"
        style="padding-left: 5px;"
      >
        test
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`thing renders component renders correctly with thing as prop in bw mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c1 {
  font-size: 1rem;
  color: #535355;
  font-style: normal;
}

.c0 {
  max-width: 50%;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: nowrap;
  -ms-flex-flow: nowrap;
  flex-flow: nowrap;
  max-height: 80%;
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
          color="#535355"
          font-size="1rem"
        >
          ☐
        </div>
        <div
          class="c1"
          color="#535355"
          font-size="1rem"
          style="padding-left: 5px;"
        >
          test
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bxivhb cqFfly"
    >
      <div
        class="sc-htpNat dzJouz"
        color="#535355"
        font-size="1rem"
      >
        ☐
      </div>
      <div
        class="sc-htpNat dzJouz"
        color="#535355"
        font-size="1rem"
        style="padding-left: 5px;"
      >
        test
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
